python/python/lance/dataset.py

```diff
@@ -595,6 +599,12 @@ def delete(self, predicate: Union[str, pa.compute.Expression]):
             predicate = str(predicate)
         self._ds.delete(predicate)
 
+    def count_deleted_rows(self) -> int:
```

Comment:
Let's make them into a set of private APIs?  

For example, `Dataset::stats()`?

Reply:
Not sure if this is what Lei means, but it would be nice to have a `dataset_stats` property that returns an object with these methods. Would group them together in a sort of namespace, making them a bit more discoverable. This is what we do with `optimize` property.

Reply:
@wjones127 could you point to the `optimize` property you're talking about? I grepped the codebase and couldn't find it.

I found this optimize module where we create an IndexRemapper that has a pointer property to the dataset. Are you thinking a similar pattern to that but with some DatasetStats type?
https://github.com/lancedb/lance/blob/fba3b4c52cf7c04e8347099781c96a6fe4e7fa65/rust/lance/src/dataset/optimize.rs#L149-L151

https://github.com/lancedb/lance/blob/fba3b4c52cf7c04e8347099781c96a6fe4e7fa65/rust/lance/src/dataset/index.rs#L34-L47

Reply:
Here:

https://github.com/lancedb/lance/blob/fba3b4c52cf7c04e8347099781c96a6fe4e7fa65/python/python/lance/dataset.py#L957-L959

And then:

https://github.com/lancedb/lance/blob/fba3b4c52cf7c04e8347099781c96a6fe4e7fa65/python/python/lance/dataset.py#L1522-L1598

Reply:
ah I see now -- sorry I was looking in the rust code. Ok this makes sense, can make this change

Reply:
Discussed with @eddyxu -- we decided a good solution would be to have the pylance Dataset expose a property called `stats` that has two methods, `index_stats` and `dataset_stats` that return all stats for index/dataset. The motivation here was to have fewer methods to call to collect all stats.